### PR TITLE
Pp 335 cherrypicker - connector pagination story (PR post cherrypicking)

### PR DESCRIPTION
**Pull request by rahul-vinayak**
## WHAT
- changed the clause of order by for find charges in design order of
  CREATION_DATE
- fixed resource and dao tests
- and renamed ChargesResource to ChargesApiResource
- Introduces a Basic test for Hal Response builder
- HalResponseBuilder class is added using which you can create a HAL
  representation for your HATEOS resource
- the pre, next, first, last, self links work according to the story
  requirement
- tests to cover the show and hide of the respective links
- default value of the display_size to be passed in from the
  configuration yaml

### CODE REVIEW

**Comment by rahul-vinayak** [comment]
The review comments from https://github.com/alphagov/pay-connector/pull/169 have been addressed in this PR

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java [hunk 1]
```diff
@@ -109,46 +102,26 @@ public Response getChargesJson(@PathParam(ACCOUNT_ID) Long accountId,
...
+                                   @QueryParam(PAGE) Long pageNumber,
+                                   @QueryParam(DISPLAY_SIZE) Long displaySize,
                                    @Context UriInfo uriInfo) {
 
         List<Pair<String, String>> inputDatePairMap = ImmutableList.of(Pair.of(FROM_DATE_KEY, fromDate), Pair.of(TO_DATE_KEY, toDate));
+        List<Pair<String, Long>> nonNegativePairMap = ImmutableList.of(Pair.of(PAGE, pageNumber), Pair.of(DISPLAY_SIZE, displaySize));
+
         return ApiValidators
-                .validateDateQueryParams(inputDatePairMap)
+                .validateQueryParams(inputDatePairMap, nonNegativePairMap) //TODO - improvement, get the entire searchparam object into the validateQueryParams
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: a builder could have been nicer here. But fine for now

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 2]
```diff
@@ -270,6 +271,44 @@ public void shouldFilterTransactionsBasedOnFromAndToDates() throws Exception {
...
+
+    @Test
+    public void shouldShowValidationsForDateAndPageDisplaySize() throws Exception {
+        ImmutableList<String> expectedList = ImmutableList.of(
+                "query param 'from_date' not in correct format",
+                "query param 'to_date' not in correct format",
+                "query param 'display_size' should be a non zero positive integer",
+                "query param 'page' should be a non zero positive integer");
+
+        ValidatableResponse response = getChargeApi
```
Comment (kakumara) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: 👍

**Comment by kakumara** [comment]
looks ok 👍

**Pull request merged**